src/collapse/perf.rs

```diff
@@ -1,123 +1,188 @@
-    ///
-    /// If disabled, the root frame is given the name of the profiled process.
-    pub include_pid: bool,
+mod logging {
+    use log::{info, warn};
 
-    /// Include TID and PID in the root frame.
-    ///
-    /// Implies `include_pid`.
-    pub include_tid: bool,
+    pub(super) fn filtering_for_events_of_type(ty: &str) {
+        info!("Filtering for events of type: {}", ty);
+    }
 
-    /// Include raw addresses (e.g., `0xbfff0836`) where symbols can't be found.
-    pub include_addrs: bool,
+    pub(super) fn weird_event_line(line: &str) {
+        warn!("Weird event line: {}", line);
+    }
 
-    /// Annotate JIT functions with a `_[j]` suffix.
+    pub(super) fn weird_stack_line(line: &str) {
+        warn!("Weird stack line: {}", line);
+    }
+}
+
+/// `perf` folder configuration options.
+#[derive(Clone, Debug)]
+pub struct Options {
+    /// Annotate JIT functions with a `_[j]` suffix. Default is `false`.
     pub annotate_jit: bool,
 
-    /// Annotate kernel functions with a `_[k]` suffix.
+    /// Annotate kernel functions with a `_[k]` suffix. Default is `false`.
     pub annotate_kernel: bool,
 
-    /// Demangle function names
+    /// Demangle function names. Default is `false`.
     pub demangle: bool,
 
-    /// Only consider samples of the given event type (see `perf list`).
-    ///
-    /// If this option is set to `None`, it will be set to the first encountered event type.
+    /// Only consider samples of the given event type (see `perf list`). If this option is
+    /// set to `None`, it will be set to the first encountered event type. Default is `None`.
     pub event_filter: Option<String>,
-}
 
-#[derive(Copy, Clone, Debug)]
-enum EventFilterState {
-    None,
-    Defaulted,
-    Warned,
+    /// Include raw addresses (e.g., `0xbfff0836`) where symbols can't be found.
+    /// Default is `false`.
+    pub include_addrs: bool,
+
+    /// Include PID in the root frame. If disabled, the root frame is given the name of the
+    /// profiled process. Default is `false`.
+    pub include_pid: bool,
+
+    /// Include TID and PID in the root frame. Implies `include_pid`. Default is `false`.
+    pub include_tid: bool,
+
+    /// The number of threads to use. Default is the number of logical cores on your machine.
+    pub nthreads: usize,
 }
 
-impl Default for EventFilterState {
+impl Default for Options {
     fn default() -> Self {
-        EventFilterState::None
+        Self {
+            annotate_jit: false,
+            annotate_kernel: false,
+            demangle: false,
+            event_filter: None,
+            include_addrs: false,
+            include_pid: false,
+            include_tid: false,
+            nthreads: *common::DEFAULT_NTHREADS,
+        }
     }
 }
 
 /// A stack collapser for the output of `perf script`.
 ///
 /// To construct one, either use `perf::Folder::default()` or create an [`Options`] and use
 /// `perf::Folder::from(options)`.
-#[derive(Default)]
+#[derive(Clone)]
```

Comment:
Same observation here as for `dtrace` -- I wonder if we need a custom `Clone` implementation that clears any transient state in the returned object.

Reply:
Going to resolve, not because issue is resolved, but because it's the same as for `dtrace`. However we resolve that issue I will resolve this one in the same way.